crates/bevy_ecs/src/storage/resource.rs

```diff
@@ -0,0 +1,168 @@
+use crate::archetype::ArchetypeComponentId;
+use crate::archetype::ArchetypeComponentInfo;
+use crate::component::{ComponentId, ComponentTicks, Components, StorageType};
+use crate::storage::{Column, SparseSet};
+use bevy_ptr::{OwningPtr, Ptr, PtrMut, UnsafeCellDeref};
+use std::cell::UnsafeCell;
+
+pub struct ResourceInfo {
```

Comment:
```suggestion
/// Metadata for [`Resource`] data stored in the [`World`].
pub struct ResourceInfo {
```

Reply:
its not just metadata it's _also_ the actual data for the resource, that `column` is basically `Option<Box<???Component??>>`

Reply:
Ah I see. Definitely needs to be renamed then.

Reply:
`ResourceData`? Open to some bikeshedding here. I suck at naming types.

Reply:
`ResourceData` works for me

Reply:
Yep, I like it.